.pipelines/cni/pipeline.yaml

```diff
@@ -12,27 +12,184 @@ stages:
+              os: linux
+            cni_dropgz_test_windows2022_amd64:
+              arch: amd64
+              name: cni-dropgz-test
+              os: windows
+              os_version: ltsc2022
+        steps:
+          - template: ../containers/container-template.yaml
+            parameters:
+              arch: $(arch)
+              name: $(name)
+              os: $(os)
+              os_version: $(os_version)
+
+  - stage: binaries
+    displayName: Build Binaries
+    dependsOn: setup
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+      demands:
+        - agent.os -equals Linux
+        - Role -equals Build
+    jobs:
+      - job:
+        displayName: "Build k8s NPM Linux Test Suite Binary"
+        steps:
+          - bash: |
+              git clone https://github.com/huntergregory/kubernetes.git --depth=1 --branch=linux-long-sleep
+              cd kubernetes
+              export PATH=$PATH:/usr/local/go/bin/
+              make WHAT=test/e2e/e2e.test
+            displayName: "Build Kubernetes e2e.test"
+          - publish: $(System.DefaultWorkingDirectory)/kubernetes/_output/local/bin/linux/amd64
+            artifact: npm_k8s
+      - job:
+        displayName: "Build k8s NPM Windows Test Suite Binary"
+        pool:
+          name: $(BUILD_POOL_NAME_DEFAULT)
+        steps:
+          - bash: |
+              git clone https://github.com/huntergregory/kubernetes.git --depth=1 --branch=sleep-before-probing
+              cd kubernetes
+              export PATH=$PATH:/usr/local/go/bin/
+              make WHAT=test/e2e/e2e.test
+            displayName: "Build Kubernetes e2e.test"
+          - publish: $(System.DefaultWorkingDirectory)/kubernetes/_output/local/bin/linux/amd64
+            artifact: npm_k8s_windows
+
+  - stage: publish
+    displayName: Publish Multiarch Manifests
+    dependsOn:
+      - build_images
+    jobs:
+      - job: manifest
+        displayName: Compile Manifests
+        pool:
+          name: "$(BUILD_POOL_NAME_DEFAULT)"
+        strategy:
+          matrix:
+            cni_dropgz_test:
+              name: cni-dropgz-test
+              platforms: linux/amd64 windows/amd64
+        steps:
+          - template: ../containers/manifest-template.yaml
+            parameters:
+              name: $(name)
+              os_versions: $(os_versions)
+              platforms: $(platforms)
+
   - template: cilium/cilium-overlay-load-test-template.yaml
     parameters:
-      name: cilium
+      name: cilium_overlay
       clusterType: cilium-overlay-up
       clusterName: "cilium-overlay"
       nodeCount: ${CILIUM_NODE_COUNT}
       vmSize: "Standard_DS4_v2"
-      dependsOn: setup
+
   - template: singletenancy/windows-cni-load-test-template.yaml
     parameters:
       name: win_cniv1
       clusterType: "windows-cniv1-up"
       clusterName: "win-cniv1"
       nodeCount: ${WINDOWS_NODE_COUNT}
-      vmSize: Standard_B2ms
-      dependsOn: setup
+      vmSize: Standard_DS4_v2

```

Comment:
any reason we are changing the vmsize ?

Reply:
NPM Tests require a larger VMSku. Was creating false negatives.